src/vis_cpu/conversions.py

```diff
@@ -1,24 +1,185 @@
+from astropy.coordinates.builtin_frames import AltAz
+from astropy.time import Time
 
 
-def lm_to_az_za(el, m):
-    """
-    Convert l and m (on intervals -1, +1) to azimuth and zenith angle.
+def enu_to_az_za(enu_e, enu_n):
+    """Convert angle cosines in ENU coordinates into azimuth and zenith angle.
+
+    For a pointing vector in ENU coordinates vec{p}, the input arguments are
+    ``enu_e = vec{p}.hat{e}`` and ``enu_n = vec{p}.hat{n}`, where ``hat{e}`` is
+    a unit vector in ENU coordinates etc.
 
     Parameters
     ----------
-    l, m : array_like
-        Normalized angular coordinates on the interval (-1, +1).
+    enu_e, enu_n : array_like
+        Normalized angle cosine coordinates on the interval (-1, +1).
 
     Returns
     -------
     az, za : array_like
         Corresponding azimuth and zenith angles (in radians).
     """
-    lsqr = el ** 2.0 + m ** 2.0
-    n = np.where(lsqr < 1.0, np.sqrt(1.0 - lsqr), 0.0)
+    lsqr = enu_n ** 2.0 + enu_e ** 2.0
+    zeta = np.where(lsqr < 1.0, np.sqrt(1.0 - lsqr), 0.0)
 
-    az = -np.arctan2(m, el)
-    za = np.pi / 2.0 - np.arcsin(n)
+    az = -np.arctan2(enu_e, enu_n)
+    za = 0.5 * np.pi - np.arcsin(zeta)
     return az, za
+
+
+def eci_to_enu_matrix(ha, lat):
+    """3x3 transformation matrix to rotate ECI to ENU coordinates.
+
+    Transformation matrix to project Earth-Centered Inertial (ECI) coordinates
+    to local observer-centric East-North-Up (ENU) coordinates at a given time
+    and location.
+
+    Note: This is a stripped-down version of the ``eq2top_m`` function.
+
+    Parameters
+    ----------
+    ha : float
+        Hour angle, in radians, where HA = LST - RA. Normally, this is passed
+        in as ``ha = -lst``.
+
+    lat : float
+        Latitude of the observer, in radians.
+
+    Returns
+    -------
+    m : array_like
+        Array of 3x3 matrices containing the rotation matrices for each time.
+    """
+    # Reference: https://gssc.esa.int/navipedia/index.php/Transformations_between_ECEF_and_ENU_coordinates
+    return np.array(
+        [
+            [np.sin(ha), np.cos(ha), 0.0 * ha],
+            [-np.sin(lat) * np.cos(ha), np.sin(lat) * np.sin(ha), np.cos(lat)],
+            [np.cos(lat) * np.cos(ha), -np.cos(lat) * np.sin(ha), np.sin(lat)],
+        ]
+    )
+
+
+def enu_to_eci_matrix(ha, lat):
+    """3x3 transformation matrix to rotate ENU to ECI coordinates.
+
+    3x3 transformation matrix to project local observer-centric East-North-Up
+    (ENU) coordinates at a given time and location to Earth-Centered Inertial
+    (ECI) coordinates.
+
+    Parameters
+    ----------
+    ha : float
+        Hour angle, in radians, where HA = LST - RA. Normally, this is passed
+        in as ``ha = -lst``.
+
+    lat : float
+        Latitude of the observer, in radians.
+
+    Returns
+    -------
+    m : array_like
+        Array of 3x3 matrices containing the rotation matrices for each time.
```

Comment:
Again, it's just one 3x3 matrix -- unless you assume ha can be an array

Reply:
I guess a good test would be to ensure that `eci_to_enu_matrix(ha, lat).dot(enu_to_eci_matrix(ha, lat) == I`.

Reply:
Thanks, I have added this test and fixed the docstring.